main.c

```diff
@@ -29,50 +26,53 @@ int main() {
             {23,21},
             {23,22,19,21}
     };
-
     while(!orderPlaced)
     {
-        switch (state)
-        {
-            case(0):
-            {
-                getUserCredentials(username, password, &state);
+        switch (state){
+            case(0):{
+                userCredentialsStep(username, password, &state);
                 break;
             }
-            case(1):
-            {
+            case(1):{
                 printFoodTypes(noOfFoodTypes, FoodType);
                 Food = getChoiceIndex(noOfFoodTypes, &state);
                 break;
             }
-            case(2):
-            {
-                printFoodOptions(noOfFoodOptions, FoodPrice, FoodType, FoodOption, Food);
+            case(2):{
+                printf("Please choose the type of %s:\n", FoodType[Food]);
+                printFoodOptions(noOfFoodOptions, FoodPrice, FoodOption, Food);
                 Type = getChoiceIndex(noOfFoodOptions[Food], &state);
                 break;
             }
-            case(3):
-            {
-                printDrinkOptions(noOfDrinks, DrinkPrice, FoodType, Drinks, Food);
+            case(3):{
+                printf("Please choose a drink to go with your %s:\n", FoodType[Food]);
+                printDrinkOptions(noOfDrinks, DrinkPrice, Drinks);
                 Drink = getChoiceIndex(noOfDrinkOptions, &state);
                 break;
             }
-            case(4):
-            {
+            case(4):{
                 printCutleryOptions();
                 Cutlery = getChoiceIndex(noOfCutleryOptions, &state);
                 break;
             }
-            case(5):
-            {
-                getAdditionalInfo(AddInfo, &state);
+            case(5):{
+                getAdditionalInfo(AddInfo);
+                state++;
                 break;
             }
-            case(6):
-            {
-                printOrderSummary(username, FoodOption, FoodPrice, Drinks, DrinkPrice, Cutlery, Food, Type, Drink, noOfDrinks);
-                printTotalandAddInfo(AddInfo, FoodPrice, DrinkPrice, Drink, Food, Type, noOfDrinks);
-                OrderConfirmation(&state, &orderPlaced);
+            case(6):{
+                printOrderHeader(username);
+                printOrderFood( FoodOption, FoodPrice, Food, Type);
+                printOrderDrinks(Drinks, DrinkPrice, Drink, noOfDrinks);
+                printOrderCutlery(Cutlery);
+                printAddInfo(AddInfo);
+                double total = 0;
+                if(Drink == noOfDrinks)
+                    total = FoodPrice[Food][Type];
+                else
+                    total = FoodPrice[Food][Type] + DrinkPrice[Drink];
+                printOrderTotal(total);
```

Comment:
Take this code and put it into a printOrder function (keep these functions but call them from printOrder, and call printOrder from main)